lib/core/src/display/shape/text/render.rs

```diff
@@ -0,0 +1,202 @@
+//! Rendering TextField.
+
+pub mod assignment;
+pub mod selection;
+
+use crate::prelude::*;
+
+use crate::display::object::DisplayObjectData;
+use crate::display::shape::glyph::font::FontRegistry;
+use crate::display::shape::glyph::font::FontRenderInfo;
+use crate::display::shape::glyph::system::GlyphSystem;
+use crate::display::shape::text::content::TextFieldContent;
+use crate::display::shape::text::content::TextFieldContentFullInfo;
+use crate::display::shape::text::cursor::Cursor;
+use crate::display::shape::text::cursor::Cursors;
+use crate::display::shape::text::render::assignment::GlyphLinesAssignment;
+use crate::display::shape::text::render::assignment::LineFragment;
+use crate::display::shape::text::render::selection::SelectionSpritesGenerator;
+use crate::display::shape::primitive::def::*;
+use crate::display::shape::primitive::system::ShapeSystem;
+use crate::display::symbol::geometry::compound::sprite::Sprite;
+
+use nalgebra::Vector2;
+use nalgebra::Vector3;
+use crate::display::shape::text::TextFieldProperties;
+use crate::display::world::World;
+
+
+// =======================
+// === RenderedContent ===
+// =======================
+
+type GlyphLine = crate::display::shape::glyph::system::Line;
+
+/// Structure containing sprites bound to one cursor with its selection.
+#[derive(Debug)]
+pub struct CursorSprites {
+    /// Cursor sprite.
+    pub cursor: Sprite,
+    /// Selection sprites.
+    pub selection: Vec<Sprite>,
+}
+
+/// Structure with all data and sprites required for rendering specific TextField.
+#[derive(Debug)]
+pub struct RenderedContent {
```

Comment:
The docs tell that this is struct with data required FOR RENDERING (in the future), while the name of the struct tells it was already RENDERED (in the past).